app/models/enterprise.rb

```diff
@@ -285,6 +285,14 @@ def distributed_taxons
       select('DISTINCT spree_taxons.*')
   end
 
+  def current_distributed_taxons
+    Spree::Taxon
+      .select("DISTINCT spree_taxons.*")
+      .joins(products: :variants_including_master)
+      .joins("INNER JOIN (#{current_exchange_variants.to_sql}) \
```

Comment:
have we checked the EXPLAIN ANALYZE of this big query? this nesting here makes it quite large.